src/dmd/expression.d

```diff
@@ -4817,13 +4817,21 @@ extern (C++) final class DotVarExp : UnaExp
                                 /* checkModify will consider that this is an initialization
                                  * of v while it is actually an assignment of a field of v
                                  */
-                                scope modifyLevel = v.checkModify(loc, sc, dve.e1, flag);
+                                scope modifyLevel = v.checkModify(loc, sc, dve.e1, flag, !onlyUnion);
                                 // reflect that assigning a field of v is not initialization of v
                                 // unless v is a (potentially nested) union
                                 if (!onlyUnion)
                                     v.ctorinit = false;
```

Comment:
Does `ctorinit` still get set to `true` here?  It looks like it doesn't because the call to `modifyFieldVar` is now elided, so should be OK to remove this as dead code.